src/Scripts.php

```diff
@@ -15,13 +15,32 @@ public function enqueueScripts() : void
         $apiKey = get_field('civic_cookie_control_api_key', 'option');
         $productType = get_field('civic_cookie_control_product_type', 'option');
         $googleAnalyticsId = get_field('google_analytics_id', 'option');
+
         if ($apiKey && $productType) {
+            // resolve customisation options to saved value or default
+            $options = new \AnalyticsWithConsent\Options;
+            $title = get_field('civic_cookie_control_title', 'option') ?? $options->getDefault('title');
+            $intro = get_field('civic_cookie_control_intro', 'option') ?? $options->getDefault('intro');
+            $necessaryDescription = get_field('civic_cookie_control_necessaryDescription', 'option') ?? $options->getDefault('necessaryDescription');
+            $analyticalDescription = get_field('civic_cookie_control_analyticalDescription', 'option') ?? $options->getDefault('analyticalDescription');
+            $closeLabel = get_field('civic_cookie_control_closeLabel', 'option') ?? $options->getDefault('closeLabel');
+            $acceptSettings = get_field('civic_cookie_control_acceptSettings', 'option') ?? $options->getDefault('acceptSettings');
+            $rejectSettings = get_field('civic_cookie_control_rejectSettings', 'option') ?? $options->getDefault('rejectSettings');
```

Comment:
I wonder if there's a way to reduce the repetition here, by having a method that  does the `get_field('civic_cookie_control_[field_name]', 'option') ?? $options->getDefault('[field_name]');` logic, and just passing it an array of field names to loop through?

Reply:
Thanks @RobjS 
Have a look what I've done now.. there's a bit of knock on effect to refactor so I'm not sure if it's better. 
When it's ok, I can sort out the tests and rebase.

Reply:
@brent-dxw That looks nicer 👍 To keep things clean (and to aid with testing) you may want to pass an instance of the `Options` class into the `Scripts` class, rather than new-ing up an instance in `Scripts::resolveOptions()`.